File: PyProject/leetcode/link_table/palindrome-linked-list.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


class Solution:
    def isPalindrome(self, head: ListNode) -> bool:
        if not head:
            return head
        quick = head
        slow = head

        while quick:
            if quick.next:
                quick = quick.next.next
                slow = slow.next
        

File: GoProject/leetcode/linklist/palindrome-linked-list.go
package linklist

func isPalindrome(head *ListNode) bool {
	if head == nil {
		return false
	}

	if head.Next == nil {
		return false
	}
	slow, quick := head, head.Next

	for slow != nil && quick != nil {
		if slow == quick {
			return true
		}
		slow = slow.Next
		if quick.Next != nil {
			quick = quick.Next.Next
		} else {
			return false
		}
	}

	return false
}
